truffle/src/main/java/org/jruby/truffle/core/encoding/EncodingManager.java

```diff
@@ -55,44 +57,46 @@ public static DynamicObject newRubyEncoding(RubyContext context, Encoding encodi
         return Layouts.ENCODING.createEncoding(context.getCoreLibrary().getEncodingFactory(), encoding, string, dummy);
     }
 
-    public DynamicObject[] getUnsafeEncodingList() {
+    public synchronized List<DynamicObject> getUnsafeEncodingList() {
```

Comment:
This synchronized is useless though :p
Since the full list is returned there could be ConcurrentModificationException when iterating while it's being modified.
So either use a CopyOnWriteArrayList here or replace with a helper that just returns one encoding and not the whole list.